src/OrleansRuntime/ReminderService/LocalReminderService.cs

```diff
@@ -202,17 +201,30 @@ public Task Stop()
         {
             if (stoppedCancellationTokenSource.IsCancellationRequested) return;
 
-            // try to retrieve reminder from all my subranges
-            myRange = ring.GetMyRange();
-            if (logger.IsVerbose2) logger.Verbose2("My range= {0}", myRange);
-            var acks = new List<Task>();
-            foreach (SingleRange range in RangeFactory.GetSubRanges(myRange))
+            using (await lockable.LockAsync())
             {
-                if (logger.IsVerbose2) logger.Verbose2("Reading rows for range {0}", range);
-                acks.Add(ReadTableAndStartTimers(range));
+                // try to retrieve reminder from all my subranges
+                var myRange = ring.GetMyRange();
+                if (logger.IsVerbose2) logger.Verbose2("My range= {0}", myRange);
+                var acks = new List<Task>();
+                var remindersNotInTable = new Dictionary<ReminderIdentity,LocalReminderData>(localReminders);
+
+                foreach (SingleRange range in RangeFactory.GetSubRanges(myRange))
+                {
+                    acks.Add(ReadTableAndStartTimers(range, remindersNotInTable));
+                }
+                await Task.WhenAll(acks);
+                foreach (var reminder in remindersNotInTable.Values)
```

Comment:
May be helpful to log at Info level the number of reminders about to be removed.